code/modules/surgery/organs/internal/ears/_ears.dm

```diff
@@ -34,28 +33,98 @@
 		to_chat(owner, span_warning("The ringing in your ears grows louder, blocking out any external noises for a moment."))
 
 	. = ..()
-	// if we have non-damage related deafness like mutations, quirks or clothing (earmuffs), don't bother processing here. Ear healing from earmuffs or chems happen elsewhere
+	// if we have non-damage related deafness like mutations, quirks or clothing (earmuffs), don't bother processing here.
+	// Ear healing from earmuffs or chems happen elsewhere
 	if(HAS_TRAIT_NOT_FROM(owner, TRAIT_DEAF, EAR_DAMAGE))
 		return
+	// no healing if failing
+	if(organ_flags & ORGAN_FAILING)
+		return
+	adjustEarDamage(0, -0.5 * seconds_per_tick)
+	if((damage > low_threshold) && SPT_PROB(damage / 60, seconds_per_tick))
+		adjustEarDamage(0, 4)
+		SEND_SOUND(owner, sound('sound/weapons/flash_ring.ogg'))
+
+/obj/item/organ/internal/ears/apply_organ_damage(damage_amount, maximum, required_organ_flag)
+	. = ..()
+	update_temp_deafness()
+
+/obj/item/organ/internal/ears/on_mob_insert(mob/living/carbon/organ_owner, special, movement_flags)
+	. = ..()
+	update_temp_deafness()
+
+/obj/item/organ/internal/ears/on_mob_remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+	UnregisterSignal(organ_owner, COMSIG_MOB_SAY)
+	REMOVE_TRAIT(organ_owner, TRAIT_DEAF, EAR_DAMAGE)
+
+/**
+ * Snowflake proc to handle temporary deafness
+ *
+ * * ddmg: Handles normal organ damage
+ * * ddeaf: Handles temporary deafness, 1 ddeaf = 2 seconds of deafness, by default (with no multiplier)
+ */
+/obj/item/organ/internal/ears/proc/adjustEarDamage(ddmg = 0, ddeaf = 0)
+	if(owner.status_flags & GODMODE)
+		update_temp_deafness()
+		return
+
+	var/mod_damage = ddmg > 0 ? (ddmg * damage_multiplier) : ddmg
+	if(mod_damage)
+		apply_organ_damage(mod_damage)
+	var/mod_deaf = ddeaf > 0 ? (ddeaf * damage_multiplier) : ddeaf
+	if(mod_deaf)
+		deaf = max(deaf + mod_deaf, 0)
+	update_temp_deafness()
+
+/// Updates status of deafness
+/obj/item/organ/internal/ears/proc/update_temp_deafness()
+	// if we're failing we always have at least some deaf stacks (and thus deafness)
+	if(organ_flags & ORGAN_FAILING)
+		deaf = max(deaf, 1 * damage_multiplier)
+
+	if(isnull(owner))
+		return
 
-	if((organ_flags & ORGAN_FAILING))
-		deaf = max(deaf, 1) // if we're failing we always have at least 1 deaf stack (and thus deafness)
-	else // only clear deaf stacks if we're not failing
-		deaf = max(deaf - (0.5 * seconds_per_tick), 0)
-		if((damage > low_threshold) && SPT_PROB(damage / 60, seconds_per_tick))
-			adjustEarDamage(0, 4)
-			SEND_SOUND(owner, sound('sound/weapons/flash_ring.ogg'))
+	if(owner.status_flags & GODMODE)
+		deaf = 0
 
 	if(deaf)
-		ADD_TRAIT(owner, TRAIT_DEAF, EAR_DAMAGE)
+		if(!HAS_TRAIT_FROM(owner, TRAIT_DEAF, EAR_DAMAGE))
+			RegisterSignal(owner, COMSIG_MOB_SAY, PROC_REF(adjust_speech))
+			ADD_TRAIT(owner, TRAIT_DEAF, EAR_DAMAGE)
 	else
 		REMOVE_TRAIT(owner, TRAIT_DEAF, EAR_DAMAGE)
+		UnregisterSignal(owner, COMSIG_MOB_SAY)
```

Comment:
actually since you're calling this a refactor would you mind changing this to be an early return now it's a bit more complicated than a `if/else` with one line in each branch

Reply:
I don't think this becomes any clearer with an early return IMO

Reply:
i just don't like the indentation but it's ultimately whatever